pkg/loghttp/entry.go

```diff
@@ -126,6 +143,8 @@ func (EntryEncoder) Encode(ptr unsafe.Pointer, stream *jsoniter.Stream) {
 	stream.WriteRaw(`"`)
 	stream.WriteMore()
 	stream.WriteStringWithHTMLEscaped(e.Line)
+	stream.WriteMore()
+	stream.WriteString(e.Labels)
```

Comment:
On a second pass, I notice this _decodes_ a json object of labels, but _encodes_  a string. This feels wrong (it will be unusual to see stringified labels in a json response but send them as an object). If we want to avoid doing the parsing until we're ready to send a response (for logproto compatibility), we could keep the representation in memory as a string, but we should definitely be able to do recursively `encode -> decode -> encode -> decode -> etc` the same data.